Dockerfile

```diff
@@ -100,7 +114,8 @@ ARG CGO_ENABLED=0
 RUN --mount=target=. --mount=target=/root/.cache,type=cache \
   --mount=target=/go/pkg/mod,type=cache \
   --mount=source=/tmp/.ldflags,target=/tmp/.ldflags,from=buildkit-version \
-  xx-go build ${GOBUILDFLAGS} -ldflags "$(cat /tmp/.ldflags) -extldflags '-static'" -tags "osusergo netgo static_build seccomp ${BUILDKITD_TAGS}" -o /usr/bin/buildkitd ./cmd/buildkitd && \
+  --mount=source=/tmp/.gcflags,target=/tmp/.gcflags,from=buildkit-version \
+  xx-go build ${GOBUILDFLAGS} -gcflags="$(cat /tmp/.gcflags)" -ldflags "$(cat /tmp/.ldflags) -extldflags '-static'" -tags "osusergo netgo static_build seccomp ${BUILDKITD_TAGS}" -o /usr/bin/buildkitd ./cmd/buildkitd && \
```

Comment:
Instead of changes in version stage I think this can just be
```
ARG BUILDKIT_DEBUG
ARG GOGCFLAGS=${BUILDKIT_DEBUG:+"-gcflags=all=-N -l"}

RUN xx-go build ${GOBUILDFLAGS} ${GOGCFLAGS}
```

Reply:
I think:

```
ARG GOGCFLAGS=${BUILDKIT_DEBUG:+"all=-N -l"}
```

Would probably be better. I think it resolves the quotation issue that this command has and it also makes sense to me. If you want to customize the `GOGCFLAGS` it feels wrong to require the person to also specify `-gcflags=`.